Dockerfile

```diff
@@ -22,20 +27,54 @@ ENV SRC=/opt/src \
 
 # e.g. stack/20200701/maxima
 ARG LIB_PATH
 
-RUN echo ${LIB_PATH?Error \$LIB_PATH is not defined}
-# Copy Libraries
-COPY ${LIB_PATH} ${LIB}
-
-# Copy optimization scripts
-COPY assets/maxima-fork.lisp assets/optimize.mac.template ${LIB_PATH}/../maximalocal.mac.template ${ASSETS}/
+# If QTYPE_STACK_REMOTE or QTYPE_STACK_COMMIT are empty, we cannot retrieve the libraries using Git and must enforce
+# LIB_PATH is set so we can copy the libraries from there.
+RUN if ([ -z "$QTYPE_STACK_REMOTE" ] || [ -z "$QTYPE_STACK_COMMIT" ]) && [ -z "$LIB_PATH" ]; then \
+    echo "\$LIB_PATH is not defined" \
+    && exit 1; \
+fi
+
+# Copy optimization scripts. If LIB_PATH was not set, the maximalocal.mac.template file will not be
+# copied and the one generated from the libraries cloned from Git will already be in place to use.
+COPY assets/maxima-fork.lisp assets/optimize.mac.template ${LIB_PATH}/../maximalocal.mac.template* ./assets/generate_maximalocal_template.php ${ASSETS}/
+
+# If QTYPE_STACK_REMOTE and QTYPE_STACK_COMMIT were both specified, we clone the repo and check
+# it out at the specified commit. We then determine the stackmaxima version, and look it up in
+# our version matrix (similar to what the buildimage.sh script does) to find the appropriate
+# Maxima and SBCL versions to use. If MAXIMA_VERSION and SBCL_VERSION were specified as build
+# args, they trump any values determined here.
+#
+# Once the libraries are cloned, and MAXIMA_VERSION and SBCL_VERSION environment variables are set
+# we copy everything in to the appropriate place, then generate the maximalocal.mac.template file
+# from the cloned libraries, also putting it in the correct place.
+#
+# Finally, we call buildscript (regardless of if QTYPE_STACK_REMOTE and QTYPE_STACK_COMMIT were set - because
+# the values can still be optained from the original build args; and if they are missing the build script
+# will throw an error) to create the environment where we will ultimately configure Maxima.
+RUN if [ -n "$QTYPE_STACK_REMOTE" ] && [ -n "$QTYPE_STACK_COMMIT" ]; then \
+    apt-get update && apt-get install -y git php \
+        && git clone ${QTYPE_STACK_REMOTE} qtype_stack && cd qtype_stack && git checkout ${QTYPE_STACK_COMMIT} && cd - \
+        && stackver=$(tail -n1 qtype_stack/stack/maxima/stackmaxima.mac | cut -d ':' -f2 | tr -d '$') \
+        && verstring=$(awk '$1 == "'"$stackver"'"{ print $0 }' versions) \
+        && export MAXIMA_VERSION=${MAXIMA_VERSION:-$(echo "$verstring" | cut -f2)} \
+        && export SBCL_VERSION=${SBCL_VERSION:-$(echo "$verstring" | cut -f3)} \
+        && mkdir -p ${LIB} \
+        && mkdir -p ${ASSETS} \
+        && cp -r qtype_stack/stack/maxima/* ${LIB} \
+        && cp qtype_stack/stack/cas/casstring.units.class.php ./ \
+        && sed -i 's/require_once/\/\/ require_once/g' casstring.units.class.php \
+        && php ${ASSETS}/generate_maximalocal_template.php > ${ASSETS}/maximalocal.mac.template; \
+fi && ./buildscript.sh
```

Comment:
Thanks @Lennart-Kramer. I've pushed a [fixup commit](https://github.com/mathinstitut/goemaxima/pull/22/commits/0a7631798a674b8534daa6e144ea6ac3033f349b) which shifts things around about such that **buildscript.sh** is run before anything to do with `LIB_PATH`. I also split it out in to a few `RUN` layers instead of just one.

Once approved by yourself I'll squash the fixup commit so you can merge.

Let me know if there's anything else you'd like fixed up :)